subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/GradleContextualExecuter.java

```diff
@@ -93,6 +93,7 @@ protected GradleExecuter configureExecuter() {
             gradleExecuter.assertCanExecute();
         } catch (AssertionError assertionError) {
             gradleExecuter = new NoDaemonGradleExecuter(getDistribution(), getTestDirectoryProvider());
+            gradleExecuter.assertCanExecute();
```

Comment:
❌ If we assert here again, we should first configure the executer, before we try to assert if it is valid.

Reply:
Yes, you're right. Thanks.